### PR TITLE
Integrate `FertilizerEvent`s into `Field`

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR integrates `FertilizerEvent`s into `Field` and adds the necessary methods for checking and executing fertilizer applications on a daily basis.

- `check_fertilizer_application()` determines if any of the `Field`s scheduled fertilizer events need to be executed that day, and if so passes them to be executed.
- `_formulate_fertilizer_required()` determines the minimum of amount of fertilizer that is need of a particular mix in order to fill the specified nutrient requirements.
- `_execute_fertilizer_application()` takes the values from a `FertilizerEvent`, determines how much fertilizer and nutrients will actually be applied, and then passes those values in the appropriate form to the `FertilizerApplication` module.
- `_record_fertilizer_application()` adds the data from individual application events to the `OutputManager`.
- All methods listed above are tested in 'test_field.py`.
- Several old placeholder methods for `Field` setup and event execution have been removed.
- `finish_line_doc.md` has been updated to reflect progress made.

### CODE REVIEW

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 1]
```diff
@@ -126,31 +127,131 @@ def _composition_sums_to_one(self) -> bool:
...
+
+        """
+        minimum_mass_for_nitrogen = (0 if nitrogen_fraction == 0 else (requested_nitrogen / nitrogen_fraction))
+        minimum_mass_for_phosphorus = (0 if phosphorus_fraction == 0 else (requested_phosphorus / phosphorus_fraction))
+
+        total_mass = max(minimum_mass_for_nitrogen, minimum_mass_for_phosphorus)
+        nitrogen_mass = total_mass * nitrogen_fraction
+        phosphorus_mass = total_mass * phosphorus_fraction
+        potassium_mass = total_mass * potassium_fraction
+        return {"mass": total_mass, "nitrogen_mass": nitrogen_mass, "phosphorus_mass": phosphorus_mass,
```
Comment (JoeWaddell) [inline SC_redesign/Crop_and_Soil/field/field.py]: could the return key `mass` be more specific, e.g. `total_mass`, `fertilizer_mass`, or `total_fertilizer_mass`?

**Review by JoeWaddell — APPROVED** [review]
Looks good to me! Nice job on documenting things in detail: and for writing concise unit tests.

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 2]
```diff
@@ -126,31 +127,131 @@ def _composition_sums_to_one(self) -> bool:
...
+        Determines the total mass of a specific fertilizer mix needed to meet the specified nutrient requirements.
+
+        Parameters
+        ----------
+        nitrogen_fraction : float
+            Fraction of fertilizer mix that is nitrogen, in range [0.0, 1.0]
+        phosphorus_fraction : float
+            Fraction of fertilizer mix that is phosphorus, in range [0.0, 1.0]
+        potassium_fraction : float
+            Fraction of fertilizer mix that is potassium, in range [0.0, 1.0]
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/field/field.py]:
```suggestion
        nitrogen_fraction : float
            Fraction of fertilizer mix that is nitrogen, in range [0.0, 1.0] (unitless)
        phosphorus_fraction : float
            Fraction of fertilizer mix that is phosphorus, in range [0.0, 1.0] (unitless)
        potassium_fraction : float
            Fraction of fertilizer mix that is potassium, in range [0.0, 1.0] (unitless)
```
Just a suggestion to maintain unit labeling as other refactored files.

File: SC_redesign/Crop_and_Soil/field/field.py [hunk 3]
```diff
@@ -126,31 +127,131 @@ def _composition_sums_to_one(self) -> bool:
...
+        mix_name : str
+            The name of the mix this fertilizer application is composed of.
+        total_mass : float
+            The total mass of phosphorus applied.
+        nitrogen_mass : float
+            The mass of nitrogen applied.
+        phosphorus_mass : float
+            The mass of phosphorus applied.
+        potassium_mass : float
+            The mass of potassium applied.
```
Comment (matthew7838) [inline SC_redesign/Crop_and_Soil/field/field.py]: Missing unit.

**Review by matthew7838 — APPROVED** [review]
Looks good, two small docstring comment only.

File: SC_redesign/Crop_and_Soil/field/field.py (hunk 1 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/field/field.py]: Good idea

File: SC_redesign/Crop_and_Soil/field/field.py (hunk 3 above)
Comment (Isgotkowitz) [reply SC_redesign/Crop_and_Soil/field/field.py]: Good catch, thanks Matthew

**Pull request merged**